sys/shell/commands/sc_udp.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_shell_commands
+ * @{
+ *
+ * @file
+ * @brief       Shell commands for using UDP
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "net/ng_netbase.h"
+#include "net/ng_udp.h"
+#include "net/ng_ipv6.h"
+
+#ifdef MODULE_NG_PKTDUMP
+static ng_netreg_entry_t _server = {NULL,
+                                    NG_NETREG_DEMUX_CTX_ALL,
+                                    KERNEL_PID_UNDEF};
+#endif
+
+
+void _send(char *addr, char *port, char *data)
+{
+    uint16_t port;
+    ng_pktsnip_t *payload, *udp, *ip;
+    ipv6_addr_t ip_addr;
+    ng_netreg_entry_t *sendto;
+
+    /* parse destination address
+     *
+     * For the future: we should read the type of address from the addr string,
+     * so figure out if IPv4 or IPv6 and build the header accordingly...
+     * So only IPv6 support for now */
+    if (_parse_addr(&ip_addr, addr) < 0) {
+        puts("Error: unable to parse destination address");
+        return;
+    }
+    /* parse port */
+    port = (uint16_t)port;
+    if (port == 0) {
+        puts("Error: unable to parse destination port");
+        return;
+    }
+
+    /* allocate payload */
+    payload = ng_pktbuf_add(NULL, data, strlen(data), NG_NETTYPE_UNDEF);
+    if (payload == NULL) {
+        puts("Error: unable to copy data to packet buffer");
+        return;
+    }
+    /* allocate UDP header */
+    udp = ng_udp_hdr_build(payload, NULL, 0, (uint8_t *)&port, 2);
```

Comment:
Just for consistency and for sake of being an example I would also prefer `ng_netreg_hdr_build()` here.